### PR TITLE
Finish rewrite

### PR DESCRIPTION
**Pull request by LukaJCB**
All tests are green and this should be pretty much on par instance wise with what we had before. The only things I deliberately removed were the instances for `R => *` and `(L, *)`. Not sure if they ever really belonged or if anyone used them.

### CODE REVIEW

File: tests/src/test/scala/cats/mtl/tests/SummonableImplicits.scala [hunk 1]
```diff
@@ -15,27 +15,24 @@ final class SummonableImplicits extends BaseSuite {
...
       def _1 = FunctorListen[WriterStrId, String]
       def _2 = FunctorListen[WriterTStrWriterTInt, String]
       def _3 = FunctorListen[WriterTStrWriterTInt, Vector[Int]]
-      def _4 = FunctorListen[WriterTStrTupleInt, Vector[Int]]
     }
 
     val FunctorRaiseTest: Unit = {
       def _1 = FunctorRaise[EitherStrId, String]
-      def _2 = FunctorRaise[EitherTStrEitherTInt, Int]
+      //def _2 = FunctorRaise[EitherTStrEitherTInt, Int]
```
Comment (LukaJCB) [inline tests/src/test/scala/cats/mtl/tests/SummonableImplicits.scala:28]: This is the only one I couldn't get to infer correctly, it should be picked up by nature of `functorRaiseForMonadPartialOrder` but for some reason it doesn't.

**Comment by LukaJCB** [comment]
Also there might be some inconsistencies in code style, I didn't have time to do a final pass yet

File: core/src/main/scala/cats/mtl/ApplicativeHandle.scala [hunk 2]
```diff
@@ -20,13 +20,155 @@ trait ApplicativeHandle[F[_], E] extends FunctorRaise[F, E] with Serializable {
 
 private[mtl] trait ApplicativeHandleInstances {
 
-  implicit def applicativeHandleForMonadError[F[_], E](implicit F: MonadError[F, E]): ApplicativeHandle[F, E] =
-    new ApplicativeHandle[F, E] {
-      def applicative = F
-      def functor = F
+  implicit final def handleEitherT[M[_], E](implicit M: Monad[M]): ApplicativeHandle[EitherTC[M, E]#l, E] = {
+    new ApplicativeHandle[EitherTC[M, E]#l, E] {
```
Comment (djspiewak) [inline core/src/main/scala/cats/mtl/ApplicativeHandle.scala:24]: Now that we have kind projector, I think it's probably worth nuking all of these partially-applied aliases. Easy search/replace post merge.

File: tests/src/test/scala/cats/mtl/tests/SummonableImplicits.scala (hunk 1 above)
Comment (djspiewak) [reply tests/src/test/scala/cats/mtl/tests/SummonableImplicits.scala:28]: Glancing at the structure, I would *guess* it runs afoul of the same bug that hits shims and forces the use of a macro replacing implicit search (namely, recursive falsification). Hard to say without looking closer though

File: tests/src/test/scala/cats/mtl/tests/Syntax.scala [hunk 3]
```diff
@@ -31,11 +31,11 @@ final class Syntax extends BaseSuite {
         FunctorListen.listens(lift)((_: String) + "suffix")
     }
     test("ApplicativeLocal") {
-      val fa: OptionT[FunctionC[String]#l, Int] = OptionT.liftF[FunctionC[String]#l, Int](_.length)
+      val fa: OptionT[Reader[String, *], Int] = OptionT.liftF[Reader[String, *], Int](Reader(_.length))
```
Comment (djspiewak) [inline tests/src/test/scala/cats/mtl/tests/Syntax.scala:34]: 👍

File: version.sbt [hunk 4]
```diff
@@ -1 +1 @@
-version in ThisBuild := "0.7.1-SNAPSHOT"
+version in ThisBuild := "0.7.2-SNAPSHOT"
```
Comment (djspiewak) [inline version.sbt:1]: *changes literally everything in the world*
*increments revision component of version*

:-)

**Review by djspiewak — APPROVED** [review]
This is *immense* and I love it. I do wish there were a way we could more systematically check that all the various permutations of instances are materialized, but the summoning tests aren't bad.

File: core/src/main/scala/cats/mtl/ApplicativeHandle.scala (hunk 2 above)
Comment (LukaJCB) [reply core/src/main/scala/cats/mtl/ApplicativeHandle.scala:24]: 100%!

File: version.sbt (hunk 4 above)
Comment (LukaJCB) [reply version.sbt:1]: ahahaha, this is actually left over from the merge with master, but it's quite hilarious nonetheless. Should we think about 1.0.0? Maybe release this early as a milestone of 1.0.0?

Comment (djspiewak) [reply version.sbt:1]: I'm all in favor of an early milestone of 1.0.0. We could be super-conservative and go with 0.8.0 if you prefer. Basically up to you.

This is definitely bringing cats-mtl more in line with what I would *expect* as a user from a 1.0 of such a library.

**Pull request merged**